src/seraphis_main/scan_balance_recovery_utils.cpp

```diff
@@ -371,25 +371,23 @@ bool try_find_legacy_enotes_in_tx(const rct::key &legacy_base_spend_pubkey,
     basic_records_in_tx_out.clear();
 
     // 1. extract enote ephemeral pubkeys from the memo
-    crypto::public_key legacy_main_enote_ephemeral_pubkey;
+    std::vector<crypto::public_key> legacy_main_enote_ephemeral_pubkeys;
     std::vector<crypto::public_key> legacy_additional_enote_ephemeral_pubkeys;
 
     extract_legacy_enote_ephemeral_pubkeys_from_tx_extra(tx_memo,
-        legacy_main_enote_ephemeral_pubkey,
+        legacy_main_enote_ephemeral_pubkeys,
         legacy_additional_enote_ephemeral_pubkeys);
 
-    // 2. check if there are a valid number of additional enote ephemeral pubkeys
-    if (legacy_additional_enote_ephemeral_pubkeys.size() > 0 &&
-        legacy_additional_enote_ephemeral_pubkeys.size() != enotes_in_tx.size())
-        return false;
-
-    // 3. scan each enote in the tx using the 'additional enote ephemeral pubkeys'
+    // 2. scan each enote in the tx using the 'additional enote ephemeral pubkeys'
     // - this step is automatically skipped if legacy_additional_enote_ephemeral_pubkeys.size() == 0
     crypto::key_derivation temp_DH_derivation;
+    std::vector<crypto::key_derivation> temp_DH_derivations;
     LegacyContextualBasicEnoteRecordV1 temp_contextual_record{};
     bool found_an_enote{false};
 
-    for (std::size_t enote_index{0}; enote_index < legacy_additional_enote_ephemeral_pubkeys.size(); ++enote_index)
+    for (std::size_t enote_index{0}; legacy_additional_enote_ephemeral_pubkeys.size() > 0 &&
```

Comment:
```suggestion
    for (std::size_t enote_index{0}; enote_index < legacy_additional_enote_ephemeral_pubkeys.size() && enote_index < enotes_in_tx.size(); ++enote_index)
```

Otherwise it can segfault if `legacy_additional_enote_ephemeral_pubkeys.size() > 0 && legacy_additional_enote_ephemeral_pubkeys.size() < enotes_in_tx.size()`

Reply:
Oopsie,
thanks for the review :)